src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperScopeManager.cs

```diff
@@ -15,6 +15,7 @@ namespace Microsoft.AspNetCore.Razor.Runtime.TagHelpers
     /// </summary>
     public class TagHelperScopeManager
     {
+        private readonly ExecutionContextPool _executionContextPool;
         private readonly Stack<TagHelperExecutionContext> _executionScopes;
```

Comment:
Suggest removing `_executionScopes`. The `ExecutionContextPool` contains _more_ information than the stack.